runtime-tools-quarkus-extension-parent/runtime/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/forms/impl/FormsStorageImpl.java

```diff
@@ -17,59 +17,163 @@
+        try {
+            URL originURL = Thread.currentThread().getContextClassLoader().getResource(FORMS_STORAGE_PATH);
+            URI originUri = originURL.toURI();
+            String newPath = originUri.getPath().replace("target/classes", "src/main/resources");
+            URI newURI = originUri.resolve(newPath);
+            formsFolder = newURI.toURL();
+            reloadFormBaseInfoList();
+        } catch (MalformedURLException e) {
+            LOGGER.warn(e.getMessage());
+        } catch (URISyntaxException e) {
+            LOGGER.warn(e.getMessage());
+        }
     }
 
     @Override
     public int getFormsCount() {
-        if (formsFolder == null) {
-            return 0;
+        return formInfoMap.keySet().size();
+    }
+
+    @Override
+    public Collection<FormInfo> getFormInfoList(FormFilter filter) {
+        final Collection<FormInfo> formInfoCollection = new ArrayList();
+        Collection<String> matchedNames = null;
+        if (filter != null && filter.getNames().size() > 0) {
+            matchedNames = formInfoMap.keySet().stream().filter(name -> filter.getNames().contains(name)).collect(Collectors.toList());
+        } else {
+            matchedNames = formInfoMap.keySet();
         }
 
-        File rootFolder = FileUtils.toFile(formsFolder);
+        matchedNames.forEach(name -> {
+            FormInfo formBaseInfos = formInfoMap.get(name);
+            formInfoCollection.add(formBaseInfos);
+        });
+        return formInfoCollection;
+    }
 
-        Collection<File> forms = readForms();
+    private FormInfo.FormType getFormType(String type) {
+        switch (type) {
+            case "html":
+                return FormInfo.FormType.HTML;
+            case "tsx":
+                return FormInfo.FormType.TSX;
+        }
+        return null;
+    }
 
-        forms.forEach(form -> {
-            System.out.println(form.getName());
-        });
+    @Override
+    public Form getFormContent(String formName) throws IOException {
+        File formFile = getFormFile(formName);
+        File formConfig = getFormConfigFile(formName);
+        String formConfiguration = "";
+        if (formConfig != null && formConfig.exists()) {
+            formConfiguration = IOUtils.toString(new FileInputStream(formConfig), StandardCharsets.UTF_8);
+        } else {
+            throw new FileNotFoundException(formName + " dose not found");
+        }
+        Form form = null;
+        if (formFile != null && formFile.exists()) {
+            form = new Form(IOUtils.toString(new FileInputStream(formFile), StandardCharsets.UTF_8), new FormConfiguration(formConfiguration), formName);
+        } else {
+            throw new FileNotFoundException(formName + "'s config file dose not found");
+        }
+        return form;
+    }
 
-        return forms.size();
+    private File getFormFile(String formName) {
+        if (formInfoMap != null) {
```

Comment:
Don't store the File in FileInfo, load the file in the same way as in getconfigFile